### ISSUE TITLE
Private npm packages generate linting errors

### ISSUE THREAD

**Issue by shawninder** [kilian-ito/atom-npm-outdated#3]
## Reproduction steps

- Make a package.json with a dependency on a private npm package, e.g. `"@my-namespace/my-package": "1.0.0"`
- EXPECT no linting errors or warnings
- OBSERVE a linting error saying `The package @my-namespace/my-package is not found on https://registry.npmjs.org`

## Discussion
Obviously it can be difficult for this package to get the latest version of a private package. Still:
- Perhaps it could look for an npm token in the current project's .npmrc (if such a file exists) and get the info anyways?
- Perhaps the error message should simply be changed to make it more obvious that this is not necessarily an error?
- Perhaps this case should be considered a warning instead?
- Perhaps private packages should be ignored altogether?

I'm not sure what the best behavior would be here, I'll leave that for you to decide. I just fear the current behavior pollutes the view of people working with private packages. I also expect a similar problem concerning private git repos...

**Comment by kilian-ito** [comment]
Hi @shawninder,

Currently the default `registry` is extracted from the `.npmrc` file if found (at project root level if the config has not be overrided). If the project does not contain `.npmrc` file, the default https://registry.npmjs.org registry will be used.

The error message displays the registry targeted to indicate that the module is not available on this specific registry.

Your issue seems to indicate that the `.npmrc` has not be found or that no default registry has been specified.
I think we can optimize the private registries support by adding the scoped package specific registry support.

### Enhencement

Currently supported `.npmrc` format:
```
registry=https://my-registry.com
```

Need to support scope specific registry:
```
registry=https://my-registry.com
@my-namespace:registry=https://my-second-registry.com
```

Regards

**Comment by kilian-ito** [comment]
I discovered an issue in project directory calcuation that prevent to target the correct project directory (associated to the editing `package.json`).

The issue is fixed in the version 0.13.0
